Move some WebKit code to WebKit framework

pvollan opened the pull request:
#### c0b5c33fb5e325554494405c1652d597c4699e60
<pre>
Move some WebKit code to WebKit framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=265324">https://bugs.webkit.org/show_bug.cgi?id=265324</a>
<a href="https://rdar.apple.com/118776213">rdar://118776213</a>

Reviewed by Brent Fulgham.

Move some WebKit code related to WebKit process extensions to WebKit framework. This change enables
us to use new assertion API to take out an assertion on the Networking process when holding locked
files, which is also included in this patch.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::setIsHoldingLockedFiles):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::aqcuireLockedFileGrant):
(WebKit::NetworkProcess::invalidateGrant):
(WebKit::NetworkProcess::hasAcquiredGrant const):
* Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift:
(NetworkingProcessExtension.handle(_:)):
* Source/WebKit/Shared/AuxiliaryProcessExtensions/WebKitProcessExtension.swift: Copied from Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift.
(WKProcessExtension.sharedInstance):
(WKNetworkingProcessExtension.handle(_:)):
(WKNetworkingProcessExtension.grant(_:name:)):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/271326@main">https://commits.webkit.org/271326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aaecd27bb12f089c1ebcfeb3f7b82b7f42267b22

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27985 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6623 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29288 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/30515 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25524 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8620 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4013 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/25288 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28252 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5374 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24045 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4630 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4811 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31204 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25586 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25488 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31071 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4817 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2984 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28882 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6353 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6721 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5261 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5305 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -236,7 +236,11 @@ WK_EXCLUDED_WEBPUSHD_SANDBOX = $(WK_EXCLUDED_WEBPUSHD_SANDBOX_$(WK_RELOCATABLE_W
-EXCLUDED_SOURCE_FILE_NAMES = $(EXCLUDED_PRODUCT_DEPENDENCY_NAMES) $(EXCLUDED_IOS_RESOURCE_FILE_NAMES) $(EXCLUDED_MACOS_PLUGIN_FILE_NAMES) $(EXCLUDED_MIGRATED_HEADERS_COCOA_TOUCH_$(WK_IS_COCOA_TOUCH)) $(WK_EXCLUDED_WEBPUSHD_SANDBOX);
+EXCLUDED_EXTENSION_SOURCE_FILE_NAMES = Shared/AuxiliaryProcessExtensions/*;
+EXCLUDED_EXTENSION_SOURCE_FILE_NAMES[sdk=iphoneos17.4.internal] = ;
+EXCLUDED_EXTENSION_SOURCE_FILE_NAMES[sdk=iphonesimulator17.4.internal] = ;
```
I'd recommend inverting this and adding some extra boilerplate to prevent this from being immediately out of date in future iOS versions.

```suggestion
EXCLUDED_EXTENSION_SOURCE_FILE_NAMES = Shared/AuxiliaryProcessExtensions/*;

// already defined further down…
INCLUDED_SOURCE_FILE_NAMES = $(INCLUDED_EXTENSION_SOURCE_FILE_NAMES_$(USE_INTERNAL_SDK));

INCLUDED_EXTENSION_SOURCE_FILE_NAMES_YES[sdk=iphone*] = Shared/AuxiliaryProcessExtensions/*;
INCLUDED_EXTENSION_SOURCE_FILE_NAMES_YES[sdk=iphone*17.0.internal] = ;
INCLUDED_EXTENSION_SOURCE_FILE_NAMES_YES[sdk=iphone*17.1.internal] = ;
INCLUDED_EXTENSION_SOURCE_FILE_NAMES_YES[sdk=iphone*17.2.internal] = ;
INCLUDED_EXTENSION_SOURCE_FILE_NAMES_YES[sdk=iphone*17.3.internal] = ;```
```

pvollan replied:
Ah, that's a good point. I have updated the patch with this change.

Thanks for reviewing!

emw-apple replied:
Sorry, I didn't mean for you to write "already defined further down…" literally. There is already a `INCLUDED_SOURCE_FILE_NAMES` definition in the file, so this change needs to be incorporated to it.

pvollan replied:
Good point, I realized that when I built locally :)

Thanks for reviewing!

emw-apple commented on Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
```diff
@@ -54,6 +54,11 @@
 #import <pal/spi/cocoa/NEFilterSourceSPI.h>
 #endif
 
+#if USE(EXTENSIONKIT) && __has_include(<WebKit-Swift.h>)
```
Why? It seems like `USE(EXTENSIONKIT)` implies that we're compiling swift code and generating the Swift header.

emw-apple commented on Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
```diff
@@ -251,4 +256,38 @@ void saveCookies(NSHTTPCookieStorage *cookieStorage, CompletionHandler<void()>&&
+#if USE(RUNNINGBOARD) && USE(EXTENSIONKIT)
+bool NetworkProcess::aqcuireLockedFileGrant()
+{
+#if __has_include(<WebKit-Swift.h>)
```
Ditto.

pvollan replied:
This was resolved by your request to invert the logic above.

emw-apple reviewed (commented):
I'm not thrilled about adding Swift code into the main WebKit target at this time, for a couple of reasons:
- There's a build time cost of turning WebKit into a three-stage compilation process. By adding Swift to the target, we have to, in sequence, compile the module interface; then the swift code; then the rest of the code.
- This change is iOS only, so it would modify the build in a way that's not uniform across the Xcode-based platforms.

emw-apple commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
```diff
@@ -28,6 +28,7 @@ CLANG_ENABLE_MODULES = YES;
 CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING = NO;
 INSTALL_PATH = $(WEBKIT_FRAMEWORKS_DIR)/WebKit.framework/$(WK_FRAMEWORK_VERSION_PREFIX)Extensions;
 SWIFT_OBJC_BRIDGING_HEADER[sdk=iphoneos17.4.internal] = Shared/AuxiliaryProcessExtensions/AuxiliaryProcessExtensionBridge.h;
+SWIFT_OBJC_BRIDGING_HEADER[sdk=iphonesimulator17.4.internal] = Shared/AuxiliaryProcessExtensions/AuxiliaryProcessExtensionBridge.h;
```
I think you need to do a similar dance here: make this setting the default, and then override it to empty on 17.0-17.3.

pvollan replied:
Fixed!

emw-apple commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
```diff
@@ -37,6 +38,7 @@ SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD = NO;
 TARGETED_DEVICE_FAMILY = 1,2,7;
 EXCLUDED_SOURCE_FILE_NAMES = *;
 EXCLUDED_SOURCE_FILE_NAMES[sdk=iphoneos17.4.internal] = ;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=iphonesimulator17.4.internal] = ;
```
Ditto. Here, it's may be to do something like:
```suggestion
EXCLUDED_SOURCE_FILE_NAMES = $(EXCLUDED_SOURCE_FILE_NAMES_$(USE_INTERNAL_SDK));;
EXCLUDED_SOURCE_FILE_NAMES_ = *;
EXCLUDED_SOURCE_FILE_NAMES_YES = *;
EXCLUDED_SOURCE_FILE_NAMES_YES[sdk=iphone*] = ;
EXCLUDED_SOURCE_FILE_NAMES_YES[sdk=iphone*17.0*] = *;
EXCLUDED_SOURCE_FILE_NAMES_YES[sdk=iphone*17.1*] = *;
EXCLUDED_SOURCE_FILE_NAMES_YES[sdk=iphone*17.2*] = *;
EXCLUDED_SOURCE_FILE_NAMES_YES[sdk=iphone*17.3*] = *;
```

pvollan replied:
Ah, good catch, will fix!

emw-apple reviewed (approved):
Per Arne and I talked and we're going to land this as-is, with a plan to remove the Swift code from the WebKit target as a follow-up.

One benefit of this is that it'll help quantify the cost of introducing Swift to the main target.

pvollan committed:
Move some WebKit code to WebKit framework
https://bugs.webkit.org/show_bug.cgi?id=265324
rdar://118776213

Reviewed by Brent Fulgham.

Move some WebKit code related to WebKit process extensions to WebKit framework. This change enables
us to use new assertion API to take out an assertion on the Networking process when holding locked
files, which is also included in this patch.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::setIsHoldingLockedFiles):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::aqcuireLockedFileGrant):
(WebKit::NetworkProcess::invalidateGrant):
(WebKit::NetworkProcess::hasAcquiredGrant const):
* Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift:
(NetworkingProcessExtension.handle(_:)):
* Source/WebKit/Shared/AuxiliaryProcessExtensions/WebKitProcessExtension.swift: Copied from Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift.
(WKProcessExtension.sharedInstance):
(WKNetworkingProcessExtension.handle(_:)):
(WKNetworkingProcessExtension.grant(_:name:)):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/271326@main
